lib/processors/manifestEnricher.js

```diff
@@ -0,0 +1,313 @@
+	const isNamespaced = namespace && resource.getPath().startsWith("/resources/" + namespace);
+
+	const bundles = getBundles(manifest);
+
+	if (bundles.length === 0) {
+		return;
+	}
+
+	await Promise.all(bundles.map(async (bundleConfig) => {
+		const {bundleUrl, bundleName, fallbackLocale} = bundleConfig;
+		let defaultBundleName = bundleUrl;
+
+		if (bundleName) {
+			defaultBundleName = bundleName.replace(/\./g, "/");
+			defaultBundleName = defaultBundleName.replace(namespace, "");
+		}
+
+		if (bundleUrl) {
+			defaultBundleName = bundleUrl.replace(".properties", "");
+
+			if (bundleUrl.startsWith("ui5://")) {
+				defaultBundleName = defaultBundleName.replace("ui5://", "").replace(namespace, "");
+			}
+		}
+
+		let pathToDefaultBundleName = "";
+
+		if (isNamespaced) {
+			pathToDefaultBundleName = path.join(`/resources/${namespace}`, defaultBundleName);
+		} else {
+			pathToDefaultBundleName = defaultBundleName.replace(namespace.replace(/\./g, "/"), "");
+		}
+		const generatedSupportedLocales = (await readdir(path.dirname(pathToDefaultBundleName)))
+			.filter((name) => name.endsWith(".properties") && name.startsWith(path.basename(defaultBundleName)))
+			.map((name) => {
+				return getLocale(name, defaultBundleName);
+			}).sort();
+
+		log.verbose("Generated supported locales: " + generatedSupportedLocales.join(", ").toString());
+
+		if (fallbackLocale && !isFallbackLocalePartOfSupportedLocales(fallbackLocale, generatedSupportedLocales)) {
+			log.error("manifest.json: Generated supported locales ('" + generatedSupportedLocales.join("', '") + "') " +
+				"not containing the defined fallback locale '" + fallbackLocale + "'. Either provide a " +
+				"properties file for defined fallbackLocale or configure another available fallbackLocale");
+			return;
+		} else if (!fallbackLocale && !isFallbackLocalePartOfSupportedLocales("en", generatedSupportedLocales)) {
+			log.warn("manifest.json: Generated supported locales ('" + generatedSupportedLocales.join("', '") + "') " +
+				"do not contain default fallback locale 'en'. Either provide a " +
+				"properties file for 'en' or configure another available fallbackLocale");
+		}
+
+		bundleConfig.supportedLocales = generatedSupportedLocales;
+	}));
+
+	resource.setString(JSON.stringify(manifest, null, options.prettyPrint ? "  " : undefined));
+	return resource;
+}
+
+
+/**
+ * @module @ui5/builder/processors/manifestEnricher
+ */
+
+/**
+ * Transforms the content of the manifest.json file.
+ *
+ * @public
+ * @function default
+ * @static
+ *
+ * @param {object} parameters Parameters
+ * @param {@ui5/fs/Resource[]} parameters.resources List of manifest.json resources to be processed
+ * @param {fs|module:@ui5/fs/fsInterface} parameters.fs Node fs or custom
+ *    [fs interface]{@link module:@ui5/fs/fsInterface}.
+ * @param {object} parameters.options Options
+ * @returns {Promise<@ui5/fs/Resource[]>} Promise resolving with the cloned resources
```

Comment:
Does it really resolve with a cloned resource? I don't see that in the code and I think it's fine if we adjust the existing resource via setString.

Reply:
done